### ISSUE TITLE
Clarification on the methodology of the paper

### ISSUE THREAD

**Issue by altria-zewei-wang** [Alibaba-NLP/ZeroSearch#22]
Dear authors, first thank you for releasing this paper, I greatly enjoyed my time reading it. Here are some questions I came up with when reading the paper. It would be greatly appreciated if you could kindly clarify them. 

1. The simulation models are finetuned based on interactions with Real Search Engines (i.e Google), but do not have access to real-time information during RL training on the policy model. As the simulation model is not finetuned on the full training set (used in RL training) but only a portion of it, **it is possible** that when generating documents in rollout, it contains incorrect information right? Since not all information are verified during rollout. 

2. During evaluation, do ZeroSearch interacts with real search engine? In other words, the informations from the interaction trajectories in Table 9 comes from real search engine, or simulation model, or generated by the policy model itself?

Thanks in advance for your time.

**Comment by sunhaonlp** [comment]
Thank you for your interest in our work!

1. **Regarding the possibility of incorrect information in generated documents during rollout:**

    Yes, it is indeed possible that the documents generated during the rollout phase may contain incorrect information. The goal of simulation tuning is to learn the text style of real search engine, rather than to memorize factual knowledge from the training data.

   To address concerns about the potential negative impact of incorrect information, we offer the following clarifications:

   - LLMs have already acquired extensive knowledge during pretraining and are **capable of generating correct information in most cases**.

   * To further enhance the model's knowledge coverage, **we incorporate a Question-Answer pair in the simulation prompt (as shown in Line 3 of Table 2).** This ensures that the LLM is aware of the ground truth for the input questions.

   * **Occasional incorrect information is acceptable.** In reality, even actual search engines can return misleading results, such as fake news or outdated information. **Our RL objective is not to have the policy model memorize retrieved content** (document masking is used to prevent this), but rather to strengthen its reasoning skills and ability to identify answers based on the presented information.

2. **Regarding the use of real search engines during evaluation:**

   Yes, during evaluation, **ZeroSearch interacts with a real search engine**. Our aim is to simulate the behavior of a search engine during training — not to replace the actual search engine at inference time. The informations from the interaction trajectories shown in Table 9 are indeed derived from real-world search engine.

We sincerely appreciate the time and attention you’ve given to reading our paper. If you have any further questions, we’d be more than happy to answer them.

**Comment by altria-zewei-wang** [comment]
Thank you very much for your immediate reply! The clarifications helps a lot. One further question arises from my side: Since I am seeing the data generated from the simulated LLM are pretty short, which may differ from real search results from platforms like Google Search. Since ZeroSearch is trained with this type of shorter simulated data, is there any post-processing step involved after getting real search engine results to fit the training data style?
